### PR TITLE
{Storage} az storage share-rm: Add process_resource_group for resource group

### PR DESCRIPTION
**Pull request by Juliehzl**
**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---
`Because resource group is not required in commandline, but it is required by SDK. In this way, we need to get resource group by ourselves when no resource group name provided.`

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S166.

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -663,7 +663,7 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
 
     for item in ['create', 'delete', 'exists', 'list', 'show', 'update']:
         with self.argument_context('storage share-rm {}'.format(item), resource_type=ResourceType.MGMT_STORAGE) as c:
-            c.argument('resource_group_name', required=False)
+            c.argument('resource_group_name', required=False, validator=process_resource_group)
             c.argument('account_name', storage_account_type)
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: storage_account_type already adds validator to parse for resource group name.

File: src/azure-cli/azure/cli/command_modules/storage/_validators.py [hunk 2]
```diff
@@ -65,13 +65,15 @@ def _cancel_timer_event_handler(_, **__):
...
 # region PARAMETER VALIDATORS
-def parse_storage_account(namespace):
+def parse_storage_account(cmd, namespace):
     """Parse storage account which can be either account name or account id"""
     from msrestazure.tools import parse_resource_id, is_valid_resource_id
 
     if namespace.account_name and is_valid_resource_id(namespace.account_name):
         namespace.resource_group_name = parse_resource_id(namespace.account_name)['resource_group']
         namespace.account_name = parse_resource_id(namespace.account_name)['name']
+    elif not namespace.resource_group_name:
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/storage/_validators.py]: what's the behavior if both account_name and resource_group_name are None?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_validators.py]: Good catch! I will add one more condition for the function although account-name is required for all commands. 😊

**Pull request merged**